Specify WHOIS and all its known numerics (#115)
ircdocs/modern-irc: _includes/modern-appendix.md

### `RPL_WHOISIDLE (317)`

>       "<client> <nick> <secs> <signon> :seconds idle, signon time

Reviewer: Missing double quote at end of message. Also applies to RPL_WHOISSPECIAL, RPL_WHOISACTUALLY and RPL_WHOISHOST.

### `RPL_ENDOFWHOIS (318)`

      "<client> <nick> :End of /WHOIS list"

Sent as a reply to the [`WHOIS`](#whois-message) command, this numeric indicates the end of a `WHOIS` response for the client with the nickname `<nick>`.

`<nick>` MUST be exactly the `<nick>` parameter sent by the client in its `WHOIS` message.
This means the case MUST be preserved, and if the client sent multiple nicks, this MUST be the comma-separated list of nicks, even if some of them were dropped.

This numeric is sent after all other `WHOIS` response numerics have been sent to the client.

### `RPL_WHOISCHANNELS (319)`

      "<client> <nick> :[prefix]<channel>{ [prefix]<channel>}

Sent as a reply to the [`WHOIS`](#whois-message) command, this numeric lists the channels that the client with the nickname `<nick>` is joined to and their status in these channels. `<prefix>` is the highest [channel membership prefix](#channel-membership-prefixes) that the client has in that channel, if the client has one. `<channel>` is the name of a channel that the client is joined to. The last parameter of this numeric is a list of `[prefix]<channel>` pairs, delimited by a SPACE character `(' ', 0x20)`.

> `RPL_WHOISCHANNELS` can be sent multiple times in the same whois reply, if the target is too many channels to fit in a single message.

Reviewer 2: ```suggestion
`RPL_WHOISCHANNELS` can be sent multiple times in the same whois reply, if the target is joined to too many channels to fit in a single message.
```

Reviewer 3: Actually I think I meant "if the target is on too many channels", which sounds better IMO

The channels in this response are affected by the [secret](#secret-channel-mode) channel mode and the [invisible](#invisible-user-mode) user mode, and may be affected by other modes depending on server software and configuration.

### `RPL_WHOISSPECIAL (320)`

      "<client> <nick> :blah blah blah

> Sent as a reply to the [`WHOIS`](#whois-message) command, this numeric is used for extra information on the client with nickname `<nick>`. This should only be used for nonessential information that does not need to be machine-readable.

Reviewer 2: ```suggestion
Sent as a reply to the [`WHOIS`](#whois-message) command, this numeric is used for extra human-readable information on the client with nickname `<nick>`. This should only be used for non-essential information that does not need to be machine-readable or understood by client software.
```

Sent to a client to inform them of the creation time of a channel. `<channel>` is the name of the channel. `<creationtime>` is a unix timestamp representing when the channel was created on the network.

### `RPL_WHOISACCOUNT (330)`

      "<client> <nick> <account> :is logged in as"

> Sent as a reply to the [`WHOIS`](#whois-message) command, this numeric indicates that the client with the nickname `<nick>` was authenticated as the owner of the `<account>`.
> 
> This does not necessarily mean the user owns their current nickname; which is covered by[`RPL_WHOISREGNICK`](#rplwhoisregnick-307).

Reviewer 2: ```suggestion
Sent as a reply to the [`WHOIS`](#whois-message) command, this numeric indicates that the client with the nickname `<nick>` was authenticated as the owner of `<account>`.

This does not necessarily mean the user owns their current nickname, which is covered by[`RPL_WHOISREGNICK`](#rplwhoisregnick-307).
```

Sent to a client to let them know who set the topic (`<nick>`) and when they set it (`<setat>` is a unix timestamp). Sent after [`RPL_TOPIC`](#rpltopic-332).

### `RPL_WHOISACTUALLY (338)`

      "<client> <nick> :is actually ...
      "<client> <nick> <host|ip> :Is actually using host"
      "<client> <nick> <username>@<hostname> <ip> :Is actually using host"

Sent as a reply to the [`WHOIS`](#whois-message) command, this numeric shows details about the client with the nickname `<nick>`.

`<username>` represents the name set by the [`USER`](#user-message) command (though `<username>` may be set by the server in other ways).

> `<host>` and `<ip>` represent the real host and IP address the client is connecting from. `<host>` CANNOT start with a colon `(':', 0x3A)` as this would get parsed as a trailing parameter – IPv6 addresses such as `"::1"` are prefixed with a zero `('0', 0x30)` to ensure this.

Reviewer 4: The 0 prefix is actually a partial expansion of the IPv6 address and it might be a good idea to mention this.

The text used in the last param of this message varies wildly.

### `RPL_WHOISSECURE (671)`

      "<client> <nick> :is using a secure connection"

> Sent as a reply to the [`WHOIS`](#whois-message) command, this numeric shows the client is using a connection to the server, considered to be reasonably safe from eavesdropping by the server (eg. connecting from localhost, using TLS, or Tor).

Reviewer 2: ```suggestion
Sent as a reply to the [`WHOIS`](#whois-message) command, this numeric shows the client is connecting to the server in a way the server considers reasonably safe from eavesdropping (e.g. connecting from localhost, using TLS, using Tor).
```
